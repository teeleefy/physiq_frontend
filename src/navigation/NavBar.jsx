import { useContext } from "react";
import {FamilyContext} from "../auth/UserContext";
import "./styles/NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";


/** Navigation bar for site. Shows up on every page.
 *
 * When user is logged in, shows links to main areas of site. When not,
 * shows link to Login and Signup forms.
 *
 * Rendered by App.
 */

function NavBar({ logout }) {
  const { currentFamily } = useContext(FamilyContext);

  function renderNavLinks(){
    if(currentFamily){
        return(
            <Nav className="ml-auto" navbar>
                <NavItem>
                <NavLink to="/home">home</NavLink>
                <NavLink to="/profile">profile</NavLink>
                <NavLink to="/" onClick={logout} id="Navbar-logout">log out</NavLink>
                </NavItem>
            </Nav> 
        )
    }
    return (
        <Nav className="ml-auto" navbar>
            <NavItem>
                <NavLink to="/login">Login</NavLink>
                <NavLink to="/signup">Sign Up</NavLink>
            </NavItem>
        </Nav>
    )
}

  return (
    <div>
        <Navbar expand="md">
            <NavLink to="/" className="navbar-brand">
            fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b>
            </NavLink>
            {renderNavLinks()}
        </Navbar>
    </div>
  );
}

export default NavBar;
