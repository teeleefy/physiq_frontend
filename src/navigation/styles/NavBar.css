nav {
    background-color: #ffffff;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
    width: 100%;
  }
  
  .nav-item {
    width: 400px;
    text-align: center;
    
  }
  
  nav a {
    color: black;
    text-decoration: none;
    margin-left: 1rem;
    font-size: 1.5rem;
  }


  
  nav a.active:not(#Navbar-logout):not(.navbar-brand) {
    font-weight: bold;
    border-bottom: 5px dotted;
    box-sizing: border-box;
  }
  
  nav a.active:not(.navbar-brand):not(.MemberNavBar-link):hover {
    color: red !important;
  }
  
  nav a:not(.MemberNavBar-link):hover {
    text-decoration: none;
    color: red;
  }
  
  nav a.active.navbar-brand {
    border-bottom: none;
  }
  
  nav a:not(.navbar-brand):hover {
    font-weight: bold;
  }
  
  .navbar-brand{
    color: black !important;
    font-weight: 600 !important;
    font-size: 2.5rem !important;
  }