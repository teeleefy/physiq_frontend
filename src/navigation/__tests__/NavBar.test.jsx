import { renderWithRouterAndFamilyContext, renderWithRouterAndLoggedOutContext, screen } from '../../../__tests__/test-tools';
import NavBar from '../NavBar'

describe('NavBar when logged out', () => {
  it('renders the NavBar component when logged out', () => {
    renderWithRouterAndLoggedOutContext(<NavBar />)
    
    screen.debug(); // prints out the jsx in the Home component unto the command line
  })

  it("matches snapshot when logged out", function() {
    const { asFragment } = renderWithRouterAndLoggedOutContext(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('NavBar with Family Context', () => {
  it('renders the NavBar component', () => {
    renderWithRouterAndFamilyContext(<NavBar />)
    
    screen.debug(); // prints out the jsx in the Home component unto the command line
  })

  it("matches snapshot", function() {
    const { asFragment } = renderWithRouterAndFamilyContext(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
})

