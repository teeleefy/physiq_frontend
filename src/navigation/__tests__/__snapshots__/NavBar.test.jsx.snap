// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar when logged out > matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          fa
          <b
            class="mychart"
          >
            m
          </b>
          il
          <b
            class="mychart"
          >
            y
          </b>
           
          <b
            class="mychart"
          >
            chart
          </b>
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
            <a
              class=""
              href="/signup"
            >
              Sign Up
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`NavBar with Family Context > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          fa
          <b
            class="mychart"
          >
            m
          </b>
          il
          <b
            class="mychart"
          >
            y
          </b>
           
          <b
            class="mychart"
          >
            chart
          </b>
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/home"
            >
              Home
            </a>
            <a
              class=""
              href="/profile"
            >
              Profile
            </a>
            <a
              aria-current="page"
              class="active"
              href="/"
              id="Navbar-logout"
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
